core/auth/fake/identifier.go

```diff
@@ -0,0 +1,236 @@
+	"flamingo.me/flamingo/v3/framework/config"
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"flamingo.me/flamingo/v3/framework/web"
+)
+
+type (
+	// Identifier is the fake Identifier implementation
+	Identifier struct {
+		responder     *web.Responder
+		broker        string
+		reverseRouter web.ReverseRouter
+		eventRouter   flamingo.EventRouter
+		config        fakeConfig
+	}
+
+	fakeConfig struct {
+		Broker           string                `json:"broker"`
+		LoginTemplate    string                `json:"loginTemplate"`
+		ValidatePassword bool                  `json:"validatePassword"`
+		UsernameFieldID  string                `json:"usernameFieldId"`
+		PasswordFieldID  string                `json:"passwordFieldId"`
+		UserConfig       map[string]userConfig `json:"userConfig"`
+	}
+
+	userConfig struct {
+		Password string `json:"password"`
+	}
+
+	viewData struct {
+		FormURL    string
+		Message    string
+		UsernameID string
+		PasswordID string
+	}
+)
+
+const (
+	defaultLoginTemplate = `
+<body>
+  <h1>Login!</h1>
+  <form name="fake-login-form" action="{{.FormURL}}" method="post">
+	<div>{{.Message}}</div>
+	<label for="{{.UsernameID}}">Username</label>
+	<input type="text" name="{{.UsernameID}}" id="{{.UsernameID}}">
+	<label for="{{.PasswordID}}">Password</label>
+    <input type="password" name="{{.PasswordID}}" id="{{.PasswordID}}">
+	<button type="submit" id="submit">Fake Login</button>
+  </form>
+</body>
+`
+
+	defaultUserNameFieldID = "username"
+	defaultPasswordFieldID = "password"
+
+	userDataSessionKey = "core.auth.fake.%s.data"
+)
+
+var (
+	_ auth.RequestIdentifier = (*Identifier)(nil)
+	_ auth.WebCallbacker     = (*Identifier)(nil)
+	_ auth.WebLogouter       = (*Identifier)(nil)
+
+	errMissingUsername  = errors.New("missing username")
+	errInvalidUser      = errors.New("invalid user")
+	errMissingPassword  = errors.New("missing password")
+	errPasswordMismatch = errors.New("password mismatch")
+)
+
+// IdentityProviderFactory -
+func IdentityProviderFactory(cfg config.Map) (auth.RequestIdentifier, error) {
+	var fakeConfig fakeConfig
+	if err := cfg.MapInto(&fakeConfig); err != nil {
+		return nil, err
+	}
+
+	return &Identifier{broker: fakeConfig.Broker, config: fakeConfig}, nil
+}
+
+// Inject injects module dependencies
+func (i *Identifier) Inject(
+	reverseRouter web.ReverseRouter,
+	responder *web.Responder,
+	eventRouter flamingo.EventRouter,
+) *Identifier {
+	i.reverseRouter = reverseRouter
+	i.responder = responder
+	i.eventRouter = eventRouter
+
+	return i
+}
+
+// Broker returns the broker id from the config
+func (i *Identifier) Broker() string {
+	return i.broker
+}
+
+// Authenticate action, fake
+func (i *Identifier) Authenticate(_ context.Context, r *web.Request) web.Result {
+	var formError error
+
+	postValues, err := r.FormAll()
+	if err == nil {
+		delete(postValues, "broker")
+		if len(postValues) > 0 {
+			formError = i.handlePostValues(r, postValues, i.broker)
```

Comment:
handlePostValues should only be called for post requests. Otherwise error messages are shown every time the page is accessed.